### PR TITLE
couple of fixes (sort lib button, keeping filter focused, ellipsis directive)

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner @yipingliao @2is10 

fixes on ellipsis directive from #11684

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js [hunk 1]
```diff
@@ -17,51 +17,64 @@ angular.module('kifi')
...
-          attributes.lastWindowResizeWidth = 0;
-          attributes.lastWindowResizeHeight = 0;
-          attributes.lastWindowTimeoutEvent = null;
+        var copyElement = element.clone();
+        copyElement.appendTo(element.parent());
+        copyElement.css('display', 'block');
+        copyElement.css('position', 'absolute');
+        copyElement.css('top', '-99999px');
+        copyElement.css('left', '-99999px');
+        copyElement.css('visibility', 'hidden');
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js]: When you have a choice, it’s generally preferable to style elements before attaching them to the document.

Also, don’t feel the need to reformat the code here for my sake, but for your own information, chaining and bulk operations are some of the best features of jQuery:

``` js
var copyElement = element.clone()
  .css({
    'display': ...,
    'position': ...,
    ...
  })
  .appendTo(element.parent());
```

File: kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js [hunk 2]
```diff
@@ -17,51 +17,64 @@ angular.module('kifi')
...
+            copyElement.html(scope.ngBind);
+            var currentHeight = copyElement.height();
             var currentNumLines = currentHeight / heightPerLine;
             var maxIndex = scope.ngBind.length;
 
-            if (currentNumLines <= 2) { // entire name fits
+            var maxNumLines =
+              (typeof(attributes.maxNumLines) !== 'undefined') ?
+                attributes.maxNumLines :
+                1;
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js]: `attributes.maxNumLines || 1` should be okay, no? zero wouldn’t make sense.

File: kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js [hunk 3]
```diff
@@ -17,51 +17,64 @@ angular.module('kifi')
...
 
       return function(scope, element, attributes) {
-        /* Window Resize Variables */
-          attributes.lastWindowResizeTime = 0;
-          attributes.lastWindowResizeWidth = 0;
-          attributes.lastWindowResizeHeight = 0;
-          attributes.lastWindowTimeoutEvent = null;
+        var copyElement = element.clone();
+        copyElement.appendTo(element.parent());
+        copyElement.css('display', 'block');
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js]: You can chain this, like `copyElement.appendTo(element.parent()).css()`, with a newline before the second dot.
Also, `.css()` takes in a JavaScript object, so you can dump everything in one call. :) (http://api.jquery.com/css/#css-properties)

**Comment by 2is10** [comment]
nice improvements, thanks

**Comment by yipingliao** [comment]
:+1:

File: kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js (hunk 3 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/common/directives/ellipsis/ellipsis.js]: nvm, this is just a subset of Jared's comment above. :)

**Comment by ahsu1230** [comment]
thanks for all the comments & help!

**Pull request merged**